app/v2/models/domain/BFLoss.scala

```diff
@@ -27,27 +27,31 @@ case class BFLoss(typeOfLoss: TypeOfLoss,
 object BFLoss {
   implicit val reads: Reads[BFLoss] = Json.reads[BFLoss]
 
-  implicit val writes: Writes[BFLoss] = (loss: BFLoss) => if (loss.typeOfLoss.isUkProperty) {
-    Json.obj(
-      "incomeSourceType" -> loss.typeOfLoss.toIncomeSourceType,
-      "taxYear" -> DesTaxYear.fromMtd(loss.taxYear).toString,
-      "broughtForwardLossAmount" -> loss.lossAmount
-    )
-  }
-  else if (loss.typeOfLoss.isForeignProperty){
-    Json.obj(
-      "incomeSourceId" -> loss.businessId,
+
+  implicit val writes: Writes[BFLoss] = (loss: BFLoss) => {
+     loss.typeOfLoss.isUkProperty match {
```

Comment:
I was thinking of having a tuple match e.g.
(loss.typeOfLoss.isUkProperty, loss.typeOfLoss.isForeignProperty) match {
  case (true, _) =>uk property
  case (_, true) => foreign property
  case (_, _) => the default SE case